node/pod.go

```diff
@@ -119,6 +119,17 @@ func podsEqual(pod1, pod2 *corev1.Pod) bool {
 
 }
 
+// podShouldEnqueue checks if two pods equal according according to podsEqual func and DeleteTimeStamp
+func podShouldEnqueue(oldPod, newPod *corev1.Pod) bool {
+	if !podsEqual(oldPod, newPod) {
+		return true
+	}
+	if !oldPod.DeletionTimestamp.Equal(newPod.DeletionTimestamp) {
```

Comment:
In the earlier patch, you checked all of object meta. Any reason for this change?

Reply:
I think Labels, Annotations, DeleteTimeStamp update should enqueue, the earlier patch check object meta for easier comparison. But I found podsEqual has check labels and annotations , so I thin this would be easier and effective.

Reply:
I agree. Why didn't you add labels and annotations here? Honestly, we should probably make the update pod callback on label and annotation change. -- that's work for a different time.

Reply:
@sargun `podsEqual` has already checked  labels and annotations.

Reply:
We should also check to make sure the deletion graceperiod is equal. and then I think it'll be good to go.

Reply:
See https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L220

Actually, it would only set when DeletionTimestamp is set, and if it changes, DeletionTimestamp must have been changed. So I think it is not necessary!

Reply:
I think it can change, let's take this example:

Timestamp | Actor | Action
00:00:00    | User  | Delete pod A, with Grace period of 30s
00:00:00   | K8s.   | Sets Deletion timestamp to 00:00:30, and graceperiod to 30s
00:00:15    | User  | Delete pod A, with Grace period of 15s
00:00:15   | K8s.   | Sets Deletion timestamp to 00:00:30, and graceperiod to 15s

In this, the Deletion Timestamp can change without the graceperiod changing.

Reply:
Yes, this is a case, but it would not affect the final delete time.

Reply:
If you think it is necessary,  I would like update this PR

Reply:
It doesn't hurt to add this check, IMHO.

Reply:
PR updated, PTAL

thanks